torch/csrc/jit/unpickler.h

```diff
@@ -19,14 +23,20 @@ class Unpickler {
   TH_DISALLOW_COPY_AND_ASSIGN(Unpickler);
 
  public:
-  // tensors inside the pickle are references to the tensor_table
+  // tensors inside the pickle are references to the tensor_table.
+  // class_resolver is to resolve strong class type, type_resolver_ is
+  // to resolve any JIT type. class_resolver and type_resolver are not merged
+  // here because some use cases need to get strong class type that
+  // type_resolver_ can not return.
```

Comment:
I don't think we should have a separate `ClassResolver` and `TypeResolver`, since they do the same job (take a type name and return the TypePtr). Instead, we should modify `StrongTypePtr` to have the `cu_` field be nullable (and enforce that if the `type_` field is a class, then `cu_` must be non-null). That way, we can merge the two resolvers.

Aside: the name of merged resolver should be `TypeResolver`, since that best describes it.

Reply:
sounds good!